### PR TITLE
Refactor and centralize opening db connections

### PR DESCRIPTION
**Pull request by katrogan**
Signed-off-by: Katrina Rogan <katroganGH@gmail.com>

# TL;DR
Refactor and centralize opening db connections. Currently flyteadmin service code, the cluster resource controller, the migrate calls and the scheduler initialization all use varying levels of copied around code to initialize the db connections. This makes introducing other gorm-supported db types very irritating.

Tested with flytesandbox.

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin
 - [x] Housekeeping

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description

- Removes the different database config structures that copy into one another and use just one struct
- Centralize the logic for opening a gorm db connection
- Set per-db specific configs

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/2154

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by katrogan** [comment]
PTAL @EngHabu @kumare3

**Review by pmahindrakar-oss — APPROVED** [review]
